### ISSUE TITLE
linting

### ISSUE THREAD

**Issue by lathonez** [BugsBunnyBR/ionic2-app-base#3]
Are we going to set up linting as part of this?

I'm in two minds about it:

* most people I've come across doing any sort of unit testing are linting their code
* demo linting set up as part of the test cycle is useful

However

* by putting any linting in here we are pushing a preference onto a tool that someone else may not have. e.g. I use tslint but I am recently aware of [ng2lint](https://www.npmjs.com/package/ng2lint)

**Comment by BugsBunnyBR** [comment]
Could lint be optional and we notes in the readme.md of how to plug your preferred lint tool?

----------------------------------------------

I am just thinking here... but maybe we could add an `--lint tslint|ng2lint`to ionic-cli to say we want to use lint in the project.

**Comment by lathonez** [comment]
Interesting. So let's have it off by default but the boilerplate config in gulpfile.js / webpack ready if people want it + documentation.

**Comment by BugsBunnyBR** [comment]
yes, it seems to be easy enough to let the developers configure it if the documentation is updated.

**Comment by BugsBunnyBR** [comment]
What would you think to be the best approach to make linting optional?
I can see some options:
1) remove the lint task from gulpfile and document how to add it;
2) create a boolean variable in the top of gulpfile and always execute the task checking the value of that variable; This idea can be extended to a flag in some configuration file like ionic.config

Any more ideas?

**Comment by lathonez** [comment]
In my setup, if you don't have a tslint.json I'm pretty sure there's no
linting? Seems straightforward if that's the case.

**Comment by BugsBunnyBR** [comment]
Well, you are right, but the reason why it works does not seems right. It works because of a configuration error (file not found), I think it could be better with a configuration flag. I will try something.

**Comment by lathonez** [comment]
OK agreed.
